dipy/reconst/dki.py

```diff
@@ -28,6 +28,460 @@
 from .base import ReconstModel
 
 
+def _positive_evals(L1, L2, L3, er=None):
+    """ Helper function that indentifies which voxels in a array have all
+    eigenvalues larger than zero
+
+    Parameters
+    ----------
+    L1 : ndarray
+        First independent variable of the integral.
+    L2 : ndarray
+        Second independent variable of the integral.
+    L3 : ndarray
+        Third independent variable of the integral.
+    er : float, optional
+        A eigenvalues is classified as larger than zero if it is larger than er
+
+    Returns
+    -------
+    ind : boolean (n,)
+        Array that marks the voxels that have all eigenvalues are larger than
+        zero.
+    """
+    if er is None:
+        er = np.finfo(L1.ravel()[0]).eps * 1e3
+
+    ind = np.logical_and(L1 > er, np.logical_and(L2 > er, L3 > er))
+  
+    return ind
+
+
+def carlson_rf(x, y, z, errtol=3e-4):
+    r""" Computes the Carlson's incomplete elliptic integral of the first kind
+    defined as:
+    
+    .. math::
+    
+        R_F = \frac{1}{2} \int_{0}^{\infty} \left [(t+x)(t+y)(t+z)  \right ]
+        ^{-\frac{1}{2}}dt
+
+    Parameters
+    ----------
+    x : ndarray (n,)
+        First independent variable of the integral.
+    y : ndarray (n,)
+        Second independent variable of the integral.
+    z : ndarray (n,)
+        Third independent variable of the integral.
+    errtol : float
+        Error tolerance. Integral is computed with relative error less in
+        magnitude than the defined value
+
+    Returns
+    -------
+    RF : ndarray (n,)
+        Value of the incomplete first order elliptic integral
+
+    Note
+    -----
+    x, y, and z have to be nonnegative and at most one of them is zero.
+
+    References
+    ----------
+    .. [1] Carlson, B.C., 1994. Numerical computation of real or complex
+           elliptic integrals. arXiv:math/9409227 [math.CA]
+    """
+    # Initialize RF
+    RF = np.zeros(len(x), dtype=x.dtype)
+
+    xn = x.copy()
+    yn = y.copy()
+    zn = z.copy()
+    An = (xn + yn + zn) / 3.0
+    Q = (3.*errtol) ** (-1/6.) * np.max([np.abs(An - xn), np.abs(An - yn),
+                                         np.abs(An - zn)], axis=0)
+    # Convergence has to be done voxel by voxel
+    for v in range(len(x)):
```

Comment:
Could you use ndindex here, so as to make these functions work on arbitrary-dimensional arrays?